### ISSUE TITLE
[FE] 10-10CG - BUG: "download PDF" link failing for majority of users

### ISSUE THREAD

**Issue by patrickvinograd** [department-of-veterans-affairs/va.gov-team#69670]
**Re-opening ticket - the issue is still [happening](http://sentry.vfs.va.gov/organizations/vsp/issues/6196/events/da605ca3c1554037b19806e45d6bae0a/events/?project=3&statsPeriod=90d)

## Issue Description
While perusing sentry I ran across [this issue](http://sentry.vfs.va.gov/organizations/vsp/issues/6196/?project=3&query=is%3Aunresolved+invalidauthenticity&statsPeriod=90d).

And confirmed in Datadog that nearly all attempts at invoking the download_pdf method result in a 403 - so users are unable to download their completed application.
![image](https://github.com/department-of-veterans-affairs/va.gov-team/assets/20694532/edc84699-0824-4dbc-b1b1-fb4763530762)

I believe the root cause is [this use of apiRequest with custom headers](https://github.com/department-of-veterans-affairs/vets-website/blob/84f0e2b7e5229030c511a05f474741faef9d0c30/src/applications/caregivers/components/ApplicationDownloadLink.jsx#L38) - it is not setting the stored `X-CSRF-Token` header value. If you search vets-website for `X-CSRF` you can see that other places where custom headers are sent are setting that value. 

After this is fixed, expected outcome is that we should see the incidence of that sentry issue and associated 403s drop to zero. 

## Front end tasks
- Ensure the back end data from API is displayed in local and review instance
- Ensure unit tests are available. 
- Ensure your code changes are covered by E2E tests
- Run axe checks using the Chrome or Firefox browser plugin

## Acceptance Criteria

- [x] Unit tests pass
- [x] E2E tests cover current code and regression
- [x] End-to-end tests show 0 violations.
- [x] The data returned matches API response (for given user or scenario)
- [ ] Continue investigation into the issue, as the initial remedy did not resolve the errors
- [ ] Once root cause is found, and solution determined, share with PM
- [ ] Once solution is in place in Staging, share with PM for validation in Sentry
- [ ] Once solution is in place in Production, share with PM for monitoring [Sentry log](http://sentry.vfs.va.gov/organizations/vsp/issues/6196/events/da605ca3c1554037b19806e45d6bae0a/events/?project=3&statsPeriod=90d)

## Appendix
<details> 

 <summary>Ensure your code changes are covered by E2E tests (expand)</summary>
 
 - Add E2E tests if none exist for this addition/change. 
 
 - Update existing E2E tests if this code will change functionality. 
 
</details>

**Comment by emily-brown34** [comment]
@jadon1979 can you check off the ACs and add any relevant notes, documents, screenshots etc to the ticket please

**Comment by emily-brown34** [comment]
and if we can throw some points on this @jadon1979

**Comment by hdjustice** [comment]
removed tasks not needed for this fix, there were no design changes and all tests pass as expected.

**Comment by hdjustice** [comment]
Reopening - the downloads still appear to be failing.  Needs further investigation

**Comment by jadon1979** [comment]
I've tested this locally and on staging and production with no errors generated.  Browsers tested were Safari, Chrome, and Edge.  I am closing this until it becomes such an issue that requires more time.
